### PR TITLE
CLOUDP-256535 Use labels to list containers

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes
Use labels to list containers

_Jira ticket:_ CLOUDP-256535

Closes #[issue number]

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by blva — APPROVED** [review]

File: internal/container/docker.go [hunk 1]
```diff
@@ -229,7 +229,7 @@ func (e *dockerImpl) ContainerList(ctx context.Context, names ...string) ([]Cont
 
 	if len(names) > 0 {
 		for _, name := range names {
-			args = append(args, "-f", "name="+name)
+			args = append(args, "-f", "label="+name)
```
Comment (tibulca) [inline internal/container/docker.go:232]:
```suggestion
			args = append(args, "-f", "label="+label)
```

File: internal/container/docker.go [hunk 2]
```diff
@@ -229,7 +229,7 @@ func (e *dockerImpl) ContainerList(ctx context.Context, names ...string) ([]Cont
 
 	if len(names) > 0 {
 		for _, name := range names {
```
Comment (tibulca) [inline internal/container/docker.go:231]:
```suggestion
		for _, label := range labels {
```
let's rename `names` to `labels` to match the filter

File: internal/podman/client.go [hunk 3]
```diff
@@ -270,7 +270,7 @@ func (o *client) RemoveContainers(ctx context.Context, names ...string) ([]byte,
 func (o *client) ListContainers(ctx context.Context, nameFilter string) ([]*Container, error) {
```
Comment (tibulca) [inline internal/podman/client.go:270]:
```suggestion
func (o *client) ListContainers(ctx context.Context, labelFilter string) ([]*Container, error) {
```

File: internal/podman/client.go [hunk 4]
```diff
@@ -270,7 +270,7 @@ func (o *client) RemoveContainers(ctx context.Context, names ...string) ([]byte,
...
 	args := []string{"ps", "--all", "--format", "json"}
 	if nameFilter != "" {
-		args = append(args, "--filter", "name="+nameFilter)
+		args = append(args, "--filter", "label="+nameFilter)
```
Comment (tibulca) [inline internal/podman/client.go:273]:
```suggestion
		args = append(args, "--filter", "label="+labelFilter)
```

**Review by tibulca — APPROVED** [review]

**Pull request merged**